pkg/aggregator/utils.go

```diff
@@ -0,0 +1,135 @@
+package aggregator
+
+import (
+	"context"
+	"runtime"
+
+	"github.com/iver-wharf/wharf-api-client-go/v2/pkg/model/request"
+	"github.com/iver-wharf/wharf-api-client-go/v2/pkg/wharfapi"
+	"github.com/iver-wharf/wharf-cmd/internal/parallel"
+	"github.com/iver-wharf/wharf-cmd/pkg/workerapi/workerclient"
+	v1 "k8s.io/api/core/v1"
+)
+
+type scopedLogStream struct {
+	wharfapi.CreateBuildLogStream
+}
+
+func newScopedLogStream(ctx context.Context, wharfapi wharfapi.Client) (*scopedLogStream, error) {
+	s, err := wharfapi.CreateBuildLogStream(ctx)
+	if err != nil {
+		return &scopedLogStream{}, err
+	}
+	scoped := &scopedLogStream{s}
+	runtime.SetFinalizer(scoped, func(s *scopedLogStream) {
+		resp, err := scoped.CloseAndRecv()
+		if err != nil {
+			log.Warn().
+				WithError(err).
+				Message("Unexpected error when closing log writer stream to wharf-api.")
+			return
+		}
+		log.Debug().
+			WithUint("inserted", resp.LogsInserted).
+			Message("Inserted logs into wharf-api.")
+	})
```

Comment:
In a bit of a coding burst I seem to have thought that it would be a good way to get rid of having to close the stream through the big `defer func(){...}()` in the `handlePod` function, and thought of it as a destructor, which it apparently isn't (when gc'd rather than instantly when out-of-scope/not referenced).

At the time the idea to extract the `func(){}()` into it's own function/method eluded me 😅